docs/notebooks/inequality_constraints.pct.py

```diff
@@ -123,13 +123,13 @@ def create_bo_model(data):
 # %% [markdown]
 # ## Run the optimization loop
 #
-# We can now run the optimization loop
+# We can now run the optimization loop. Since we have multiple datasets (objective and constraint), we use `optimize_multi` rather than `optimize`.
 
 # %%
 num_steps = 20
 bo = trieste.bayesian_optimizer.BayesianOptimizer(observer, search_space)
 
-data = bo.optimize(
+data = bo.optimize_multi(
```

Comment:
Is there a way of automatically getting this behaviour, without having to remember to do this? Or perhaps just get intuitive error messages saying to use optimize_mulit?

Reply:
Done. Note that AcquisitionRule still has both acquire and acquire_multi as combining those would mean every new AcquisitionRule would have to explicitly handle the single model case. I've added an error message to tell people to use acquire_multi if they accidentally pass in a tagged model or dataset.

Reply:
LGTM